onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -1040,7 +1041,14 @@ Status QnnBackendManager::ExtractBackendProfilingInfo() {
   const QnnProfile_EventId_t* profile_events{nullptr};
   uint32_t num_events{0};
   Qnn_ErrorHandle_t result = qnn_interface_.profileGetEvents(profile_backend_handle_, &profile_events, &num_events);
-  ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != result, "Failed to get profile events. Error: ", QnnErrorHandleToString(result));
+  if (!qnn_saver_path_.empty()) {  // Using QNN Saver backend
+    // QNN SDK 2.28.2 returns QNN_SAVER_ERROR_DUMMY_RETVALUE, but previous QNN versions return QNN_PROFILE_NO_ERROR.
+    // We accept both values.
+    ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != result && QNN_SAVER_ERROR_DUMMY_RETVALUE != result,
```

Comment:
NOTE: The `QNN_SAVER_ERROR_DUMMY_RETVALUE` enum has been available for QNN sdk versions as far back as 2.23 (maybe older). The QNN header states that this error code is returned from APIs where Saver has no capability to fulfill the request but can still record the API call in saver_output.c. Apparently, this value has always been returned from APIs like `QnnProfile_getEventData()`, but now it is also returned from `profileGetEvents()`.